### PR TITLE
fix wrong path conversion

### PR DESCRIPTION
**Pull request by jcppman**
# Problem

when fromBase or toBase contain backtracks, result of path conversion will
be incoorect.

e.g:

when basePath is `src/app`, in js file `src/util/util.js` we
require `../style.css`:

**util.js**

```
  define(['../style.less'], function () {});
```

the css file contains a reference to an img or something:

**style.css**

```
  .foo{
    background-image: url(../img/bg.png);
  }
```

when building the whole app as one .js, after normalizing, in the js the url
is incorrect. Because outPath is `baseUrl + filePath` in this case
`src/app/../util/util.js` and in `absoluteURI()` it treat `'..'` as normal
path part, which cause the wrong result.
# Solution

"normalize" `fromBase` and `toBase` in the very begining of `normalizeCSS` function.

A test is added to reveal the bug on the edge case, it will be failed before applying
the patch, and will pass after applying it.

### CODE REVIEW

**Comment by alundiak** [comment]
@jcppman Ideally, one change, should be consistent and be contained in one commit. So would be great to squash commits. But here, is use case, when real fix and test are not business related. Anyway, I would really suggest to squash. Also please change commit message `add a test that reveal a bug` to some valuable and mention instead of `that` the ID of bug, or text description. Looking to git log/history, it will not be clear, what exactly "that" bug is meant.

**Comment by jcppman** [comment]
this PR is re-organized, sorry for the confusion, it should be clear and straight forward now.

**Comment by alundiak** [comment]
@jcppman no worries. I will review PR again, but FYI, I've recently joined to this prj, soI need some time to get into the codebase, for proper and quick review and as maintainer taking responsibility for well coded feature and well tested code :)

File: normalize.js [hunk 1]
```diff
@@ -111,11 +126,8 @@ define(function() {
     
     return out.substr(0, out.length - 1);
   };
-  
-  var normalizeCSS = function(source, fromBase, toBase) {
 
-    fromBase = removeDoubleSlashes(fromBase);
-    toBase = removeDoubleSlashes(toBase);
+  var normalizeCSS = function(source, fromBase, toBase) {
```
Comment (alundiak) [inline normalize.js]: I see, that `var normalizeCSS` line is not proper aligned (1 space needed). Please update/amend/squash commit.

Comment (jcppman) [inline normalize.js]: what do you mean?

It is on second indent level (inside the first-level `define(function(){`) so would be two spaces, and there are two spaces before `var normalizeCSS`.

Comment (alundiak) [inline normalize.js]: alignment

Comment (jcppman) [inline normalize.js]: The function `relativeURI` has already ended at line 128, `var normalizeCSS = function () {` is at the same indent level as `relativeURI`, and line 132, 133, 135 etc is **inside** the function which make them one more indent level than `var normalizeCSS`.

screen shot 2017-02-20 at 11 15 45 am

Comment (alundiak) [inline normalize.js]: @jcppman OMG, this is so my bad :) Thanks. I'm going to merge this PR.

But before that, could u please let me know what test u was relying on during this fix?

Comment (alundiak) [inline normalize.js]: ok, I checked ur code locally, and test are fine. All tests using `convertURIBase()` passed.
